domain/modelconfig/service/service.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/juju/errors"
+	"github.com/juju/schema"
+
+	"github.com/juju/juju/cloud"
+	"github.com/juju/juju/environs/cloudspec"
+	"github.com/juju/juju/environs/config"
+)
+
+type CloudService interface {
```

Comment:
not sure if we want to put these interfaces in an `interface.go` file like we've been doing elsewhere

Reply:
Yeah I think that is not a bad idea.